llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -8422,6 +8422,10 @@ TargetLowering::createSelectForFMINNUM_FMAXNUM(SDNode *Node,
           Opcode == ISD::STRICT_FMINNUM || Opcode == ISD::STRICT_FMAXNUM) &&
          "Wrong opcode");
 
+  EVT VT = Node->getValueType(0);
+  if (VT.isVector() && isOperationLegal(Opcode, VT.getScalarType()))
+    return SDValue();
```

Comment:
This doesn't fit with the name of the function, so does something like this belong in the caller?

Usually this type of decision is made on what is legal for the vector type, not the scalar